wiki/Beatmap/Pattern/osu!mania/Stream/en.md

```diff
@@ -0,0 +1,39 @@
+---
+stub: true
+---
+
+# Stream
+
+*For specific stream types unrelated to 4-key, refer to [Chordstream](/wiki/Beatmap/Pattern/osu!mania/Chordstream).*
+
+**Streams** are constant notes within the same snap interval that occur for at least one beat, avoiding an occurance of a [minijack](/wiki/Beatmap/Pattern/osu!mania/Jack#minijack). There are many different types of streams that impose different levels of strain on the player. When a stream is used on its own, it's likely referring to a **single note stream**.
+
+![](img/stream.png "This image shows a specific type of stream known as a roll.")
+
+A **roll** is a common variation of a stream consisting of notes that go from the first to the last column or last to the first column.
+
+**Streams** that appear shorter than one beat are typically referred to as **bursts**, which are a shorter and quick succession of notes. They commonly use snaps higher than 1/4th.
+
+![](img/burst.png)
+
+## Jumpstream
```

Comment:
if all of the below are types of chordstreams, then why aren't they just put into the chordstream article? or rather, why is "chordstream" a separate article, when it is a type of stream?

Reply:
I debated this before but I think they should be separated due to the term chordstream never being used in 4k and instead terms like jumpstream and handstream (omitting quadstream since it's a 4k exclusive term) are used, whilst within 7k this term isn't used as much and instead chordstreams are favoured. This is mostly because of how much of an impact the difference in densities can make on patterning in 4k due to it's limited columns but also since 7k is much more likely to go beyond chordsizes of 3 more frequently. 

The wiki is also _kind of_ moreso centred around 4k patterns since it's the most popular mode by quite a big margin, so I though it would make sense to keep the terms exclusive to 7k separated to avoid confusion along with give people who are looking for that specific terminology a more optimal place to reference back to instead of one long article page in which half of the information is practically irrelevant but technically still applies.

Reply:
i disagree with separating 4k and 7k terms

- if anything, wouldn't it be "4k" and "other key modes"? but that makes even less sense and way too arbitrary. there's no kind of categorisation based on popularity anywhere else on the wiki, as that would be biased
- it's not like 4k terms aren't used in 7k and other key modes?
- it would be better to just explain which terms are used in what contexts

Reply:
i disagree with separating 4k and 7k terms

- if anything, wouldn't it be "4k" and "other key modes"? but that makes even less sense and way too arbitrary. there's no kind of categorisation based on popularity anywhere else on the wiki, as that would be biased
- it's not like 4k terms aren't used in 7k and other key modes?
- it would be better to just explain which terms are used in what contexts

Reply:
"Other keymodes" was already initially referred to though I see it was changed already, it doesn't really seem that arbitrary to me considering that is how osu!mania is mainly divided (i.e. global leaderboards being split into 4k/7k rankings, RC having exclusive guidelines for both 4k and 7k, etc.). This is simply the basis for how osu!mania has operated for years already because they keymodes are so different and 4k is statistically the more populated mode by a big margin.

**In the case of streams** (important because these are some of the only terms like this), 4k terms like the aforementioned are not really, if not never used in 7k. Again, terms like jumpstream, handstream, and chordstream stem from different keymodes for different reasons entirely and so their meaning is not necessarily building on each other but rather distinct, and so it's not easily interchangable. So, although their literal definitions are technically applicable, they just don't make any sense to practically use among the 7k community.

I really think it would just make more sense to keep them separated to avoid all the technicalities and overlap altogether, having the chordstream page just mention that these terms are exclusive to modes outside of 4k would align moreso with how these definitions function practically.

Reply:
then let's keep them separate but have them in the same article like this. what do you think?

having a separate adjacent article for a subtype is just too much jank

```markdown
# Stream

## Single-note stream

General term

## Jumpstream

4K term

## Handstream

4K term

## Quadstream

4K term

## Chordstream

Non-4K term
```

Reply:
Sure, seems reasonable. I applied it now so let me know if there's anything that should be fixed